src/javascript/crypto/e2e/openpgp/pgpmime/mimenode.js

```diff
@@ -0,0 +1,198 @@
+        } else {
+          this.setHeader_(header.name, header.value);
+        }
+      }
+    }, this));
+  }
+
+  this.setBoundary_();
+};
+
+
+/**
+ * Sets the MIME message boundary for a node.
+ * @private
+ */
+e2e.openpgp.pgpmime.MimeNode.prototype.setBoundary_ = function() {
+  // TODO: Strictly ensure that the boundary value doesn't coincide with
+  //   any string in the email content and headers.
+  this.boundary_ = '---' + goog.string.getRandomString() +
+      Math.floor(Date.now() / 1000).toString();
+};
+
+
+/**
+ * Adds a child to a MIME node.
+ * @param {{
+ *   multipart: boolean,
+ *   optionalHeaders: (Array<{name: string, value: string}>|undefined)
+ * }} options Options to initialize for the node.
+ * @param {string=} opt_filename Name of the file, if one exists.
+ * @return {e2e.openpgp.pgpmime.MimeNode}
+ */
+e2e.openpgp.pgpmime.MimeNode.prototype.addChild =
+    function(options, opt_filename) {
+  var node = new e2e.openpgp.pgpmime.MimeNode(options, this, opt_filename);
+  this.children_.push(node);
+  return node;
+};
+
+
+/**
+ * Sets a MIME header.
+ * @param {string} key Name of the header.
+ * @param {string} value Value of the header, possibly including parameters.
+ * @param {Object.<string, string>=} opt_params Optional dict of additional
+ *   parameters
+ * @private
+ */
+e2e.openpgp.pgpmime.MimeNode.prototype.setHeader_ =
+    function(key, value, opt_params) {
+  var headerValue = e2e.openpgp.pgpmime.Utils.parseHeaderValue(value);
+  if (opt_params) {
+    headerValue.params = headerValue.params || {};
+    goog.object.extend(headerValue.params, opt_params);
+  }
+  goog.object.set(this.header_, key, headerValue);
+};
+
+
+/**
+ * Sets the content.
+ * @param {(string|!e2e.ByteArray)} content The content to set
+ */
+e2e.openpgp.pgpmime.MimeNode.prototype.setContent = function(content) {
+  this.content_ = content;
+};
+
+
+/**
+ * Builds an RFC 2822 message from the node and its children.
+ * @return {string}
+ */
+e2e.openpgp.pgpmime.MimeNode.prototype.buildMessage = function() {
+  var lines = [];
+  var transferEncoding = this.header_[constants.Mime.CONTENT_TRANSFER_ENCODING];
+  var contentType = this.header_[constants.Mime.CONTENT_TYPE];
+
+  // Set required header fields
+  if (this.filename && !this.header_[constants.Mime.CONTENT_DISPOSITION]) {
+    // Set the correct content disposition header for attachments.
+    this.setHeader_(constants.Mime.CONTENT_DISPOSITION,
+        constants.Mime.ATTACHMENT, {filename: this.filename});
+  } else if (this.content_ && goog.typeOf(this.content_) === 'string') {
+    // sets the charset to utf-8 if the charset hasn't already been specified
+    contentType.params.charset =
+        contentType.params.charset || constants.Mime.UTF8;
+  }
+  if (this.multipart_) {
+    // Multipart messages need to specify a boundary
+    contentType.params.boundary = this.boundary_;
+  }
+
+  // TODO: Wrap header lines at 76 chars
```

Comment:
You might as well fix this TODO, as recommended by https://tools.ietf.org/html/rfc2822#section-2.2.3. `prettyTextWrap` in extension/utils/text.js should do the trick, though you'll have to refactor so that this library doesn't depend on the extension.